ui/job-view/PushHeader.jsx

```diff
@@ -140,21 +138,22 @@ export default class PushHeader extends React.PureComponent {
     }
   }
 
-  cancelAllJobs() {
-    const { repoName, revision, isLoggedIn, pushId } = this.props;
+  async cancelAllJobs() {
+    if (window.confirm('This will cancel all pending and running jobs for this push. It cannot be undone!. Are you sure?')) {
+      const { push, isLoggedIn } = this.props;
 
-    this.setState({ showConfirmCancelAll: false });
-    if (!isLoggedIn) return;
+      if (!isLoggedIn) return;
+
+      const result = await ((await PushModel.cancelAll(push.id)));
```

Comment:
I have to admit, I don't understand what the double ``await`` and extra parentheses are for.  Can you help me understand?  I've just never seen this, curious what it's about.  :)

Reply:
To have access to the result using the `fetch` API, we need to unwrap two promises. Also, 
from https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch:

> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

`!result.ok` in our case would be able to tell us if there is a failure.

The extra parenthesis should be removed :)